pandas/core/groupby.py

```diff
@@ -3210,7 +3210,15 @@ def _wrap_applied_output(self, keys, values, not_indexed_same=False):
 
         key_names = self.grouper.names
 
-        if isinstance(values[0], DataFrame):
+        # GH12824.
+        v = values[0]
+        if v is None:
```

Comment:
make this a helper function in ``core/groupby.py`` (and consolidate the other one which has the same code)